react/ButtonAction/index.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import styles from './styles.styl'
+import Icon from '../Icon'
+
+const usableByIcon = icon => {
+  const isSvgSymbol = icon && !!icon.id
+  const isIconIdentifier = typeof icon === 'string'
+  return isSvgSymbol || isIconIdentifier
+}
+
+const ButtonAction = props => {
+  const { type, disabled, children, label, leftIcon, rightIcon, compact, className, onClick, ...restProps } = props
+  return (
+    <button
+      disabled={disabled}
+      type="button"
+      role="button"
+      className={cx(
+        styles['c-actionbtn'], {
+          [styles[`c-actionbtn--${type}`]] : type,
+          [styles[`c-actionbtn--compact`]] : compact
+        },
+        className)}
+      onClick={onClick}
+      title={compact ? label : undefined}
+      {...restProps}
+    >
+      <span>
+        {usableByIcon(leftIcon) ? <Icon icon={leftIcon} /> : leftIcon}
+        {label &&
+          <span
+            data-action="label"
+            className={styles['c-actionbtn-label']}>{label}
+          </span>}
+        {rightIcon &&
+          <span
+            data-action="icon"
+            className={styles['c-actionbtn-icon']}>
+            {usableByIcon(rightIcon) ? <Icon icon={rightIcon} /> : rightIcon}
+          </span>
+        }
+        {children}
+      </span>
+    </button>
+  )
+}
+
+ButtonAction.propTypes = {
+  type: PropTypes.string.isRequired,
```

Comment:
a `oneOf` could be used here.

Since `type` has a default, it is not really required here, ie the user does not have to provide it.